### PR TITLE
Update Golang version to 1.17

### PR DESCRIPTION
**Pull request by emlaver**
## PR summary

Fixes: #242 

**Note: An existing issue is [required](https://github.com/IBM/cloudant-go-sdk/blob/master/CONTRIBUTING.md#PRs) before opening a PR.**

## PR Checklist

Please make sure that your PR fulfills the following requirements:

- [x] The commit message follows the
[Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type  

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [x] Build/CI related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

## What is the current behavior?
Golang minimum version is 1.16 

## What is the new behavior?
Update minimum supported Golang version to 1.17

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: .secrets.baseline [hunk 1]
```diff
@@ -3,7 +3,7 @@
     "files": "stubs/.+\\.json|go.sum|^.secrets.baseline$",
     "lines": null
   },
-  "generated_at": "2021-12-22T14:20:39Z",
+  "generated_at": "2022-02-04T14:56:37Z",
```
Comment (mojito317) [inline .secrets.baseline:6]: If none has changed the baseline file, we do not need to regenerate it.

**Review by mojito317 — APPROVED** [review]

**Comment by eiri** [comment]
Technically this works, but let's wait with merging till 1.18 officially released https://tip.golang.org/doc/go1.18

**Review by eiri — APPROVED** [review]
cool, but maybe revert silly `.secrets.baseline` file, it doesn't seems to have meaningful changes?

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -21,7 +21,7 @@ assignees: ''
 
 **Must gather (please complete the following information):**
  - SDK Version [e.g. 1.2.1]
- - Go Version [e.g. Go 1.16.8]
+ - Go Version [e.g. Go 1.17]
```
Comment (mojito317) [inline .github/ISSUE_TEMPLATE/bug_report.md:24]: If you change this file here, it's going to be reverted on the next generated PR. Please do this change in the generator repo.

Comment (emlaver) [reply .github/ISSUE_TEMPLATE/bug_report.md:24]: Opened PR: https://github.ibm.com/cloudant/sdks/pull/487

Comment (emlaver) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Reverted in [8110401](https://github.com/IBM/cloudant-go-sdk/pull/245/commits/8110401e5fa063106266677b7c303ed4998de8c9)

**Review by mojito317 — APPROVED** [review]
I approve this in advance the fix of https://github.com/IBM/cloudant-go-sdk/pull/245/files#r834610384

**Pull request merged**